tutorial/list-with-data.md

```diff
@@ -0,0 +1,96 @@
+---
+layout: page
+title: List of struct and ListWithData widget
+---
+
+
+Sometime, you need to display a list of complex item, more complex than a list of string, but you also want to rely on Fyne's databinding. It can be the case when you want to display a list of contact, or books for example.
+
+
+## Defining the binding
+
+
+Well first thing first, let's start by defining our binding in our app : 
+
+```go
+var ContactListBinding binding.UntypedList = binding.NewUntypedList()
+```
+
+In Fyne, the only way to bind with a slice of struct is by using a `binding.UntypedList`.
+
+## Creating the list
+
+Fyne comes with a long list of widget, but the one we want to use is ListWithData. The constructor takes three parameters : 
+
+- A databinding
+- A function that create the shape of our list element
+- A function to update our list element with the data
+
+
+```go
+widget.NewListWithData(
+		ContactListBinding,
+		func() fyne.CanvasObject {
+			return container.NewVBox(widget.NewLabel(""), widget.NewLabel(""), widget.NewLabel(""), widget.NewLabel(""))
+		},
+		func(di binding.DataItem, co fyne.CanvasObject) {
+			v, _ := di.(binding.Untyped).Get()
```

Comment:
I wonder if it is worth using `s := binding.BindStruct(v.(*contact))`.
Then a label could be bound using `Label.Bind(s.GetItem("FirstName").(binding.String))` I think?...
It would remove the code where you set up new strings and then bind them - making a proper connection to the source data.